### ISSUE TITLE
Unable to close client when using ldd

### ISSUE THREAD

**Issue by aaricpittman** [launchdarkly/ruby-server-sdk#234]
## Description
If you configure the client to `use_ldd` then you are unable to close the client. Calling `client.close` results in a `NoMethodError` for `@data_source` as it does not get defined when [using ldd](https://github.com/launchdarkly/ruby-server-sdk/blob/1b2ba7bfd9ebfccfa7717cbd5c32665bc5f380db/lib/ldclient-rb/ldclient.rb#L94-L97). 

```ruby
undefined method `stop' for nil:NilClass (NoMethodError)
```

It may be that closing the client is not necessary when using ldd, but I did not see anything in the docs that indicated that.

### Example
```ruby
config = LaunchDarkly::Config.new(
  feature_store:  LaunchDarkly::Integrations::Redis.new_feature_store(...),
  use_ldd: true
)
client = LaunchDarkly::Client.new(sdk_key, config)
client.close
```

**Expected behavior**
An exception is not thrown when calling `client.close`.

**SDK version**
`7.2.0`

**Comment by keelerm84** [comment]
Thank you for bringing this to our attention. I will look into this as soon as I am able. Once a fix has been released, I will report back here.

Tracking internally as sc-227874.

Thank you again.

**Comment by keelerm84** [comment]
This has been fixed in v7.3.2.

**Comment by aaricpittman** [comment]
@keelerm84 Thanks 🙏.
